portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,35 +24,81 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comment data. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private ArrayList<String> comments;
-
-  @Override
-  public void init() {
-    comments = new ArrayList<String>();
-  }
-
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String userComment = request.getParameter("text-input");
-    if (userComment == null) {
-      userComment = "";
+    if (userComment != null) {
+      storeComment(userComment);
     }
-    comments.add(userComment);
     response.sendRedirect("/index.html");
   }
 
+  /**
+    * Stores user comment as entity in datastore.
+    */
+  private void storeComment(String userComment) {
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", userComment);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    PreparedQuery results = getAllComments();
+    
+    ArrayList<String> comments = new ArrayList<String>();
+    int numberToDisplay = getNumberToDisplay(request); 
+    int numberDisplayed = 0;
+
+    for(Entity comment : results.asIterable()) {
+      String commentText = (String) comment.getProperty("text");
+      comments.add(commentText);
+      numberDisplayed += 1;
+      if (numberDisplayed == numberToDisplay) {
+        break;
+      }
+    }
+
     String json = convertToJson(comments); 
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
+  /** 
+    * Returns number of comments to display, based on user's 
+    * selection. 
+    */
+  private int getNumberToDisplay(HttpServletRequest request) {
+    String parameterString = (String) request.getParameter("numberToDisplay");
+    int numberToDisplay;
+    try {
+      numberToDisplay = Integer.parseInt(parameterString);
+    } catch (Exception e) {
```

Comment:
Consider only catching the exception that you mean to. E.g. NumberFormatException

Reply:
Good catch! (so to speak). I've fixed it :)